snapcraft/internal/meta.py

```diff
@@ -458,3 +461,14 @@ def _validate_hook(hook_path):
     if not os.stat(hook_path).st_mode & stat.S_IEXEC:
         asset = os.path.basename(hook_path)
         raise CommandError('hook {!r} is not executable'.format(asset))
+
+
+def _verify_app_paths(basedir, apps):
+    for app in apps:
+        path_entries = [i for i in ('desktop', 'completer')
```

Comment:
oh, this would be part of a bigger refactor @elopio might be working on as he has shown interest in refactoring the `meta` module.